### PR TITLE
Fix prometheus port (#1078) and add unit test for opentelemetry (#1088)

### PR DESCRIPTION
**Pull request by tiationg-kho**
Fix prometheus port (#1078) and add unit test for opentelemetry (#1088)

**Issue #, if available: 
#1078
#1088

**Description of changes:**
- When enablePrometheusServer config is false, we will not open the port
- Add unit test for pkg/observability/opentelemetry.go file

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/observability/opentelemetry_test.go [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+	conn, err := net.DialTimeout("tcp", fmt.Sprintf("localhost:%d", mockDefaultPort), time.Second)
+	if err != nil {
+		t.Errorf("server not listening on port %d: %v", mockDefaultPort, err)
+	}
+	conn.Close()
+
+	conn, err = net.DialTimeout("tcp", fmt.Sprintf("localhost:%d", mockClosedPort), time.Second)
+	if err == nil {
+		conn.Close()
+		t.Errorf("server should not listening on port %d: %v", mockClosedPort, err)
```
Comment (Lu-David) [inline pkg/observability/opentelemetry_test.go:162]: Nit: be

File: pkg/observability/opentelemetry_test.go [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+}
+
+func TestNodeActionsInc(t *testing.T) {
+	metrics := getMetrics(t)
+
+	metrics.NodeActionsInc(mockAction, mockNodeName1, mockEventID1, nil)
+	metrics.NodeActionsInc(mockAction, mockNodeName2, mockEventID2, nil)
+	metrics.NodeActionsInc(mockAction, mockNodeName3, mockEventID3, errors.New("mockError"))
+
+	rr := mockMetricsRequest()
```
Comment (Lu-David) [inline pkg/observability/opentelemetry_test.go:97]: What does rr mean? Could we rename to something more specific?

Comment (Lu-David) [reply pkg/observability/opentelemetry_test.go:97]: I assume request response?

File: pkg/observability/opentelemetry_test.go [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+}
+
+func TestRegisterMetricsWith(t *testing.T) {
+	const errorEventMetricsTotal = 23
+	const successActionMetricsTotal = 31
+	const errorActionMetricsTotal = 97
+
+	metrics := getMetrics(t)
+
+	errorEventLables := []attribute.KeyValue{labelEventErrorWhereKey.String(mockErrorEvent)}
```
Comment (Lu-David) [inline pkg/observability/opentelemetry_test.go:115]: Typo: errorEventLabels. Same for other areas

File: pkg/observability/opentelemetry_test.go [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+	handler.ServeHTTP(rr, req)
+	return rr
+}
+
+func validateStatus(t *testing.T, rr *httptest.ResponseRecorder) {
+	status := rr.Code
+	h.Equals(t, http.StatusOK, status)
+}
+
+func getMetricsMap(body string) map[string]string {
```
Comment (Lu-David) [inline pkg/observability/opentelemetry_test.go:202]: Could you add a comment here indicating an example of what the body might look like? Is there only a specific format it accepts? Just so it's easier to understand what this does?

**Review by Lu-David — APPROVED** [review]
Overall lgtm! Just some small comments

File: pkg/observability/opentelemetry_test.go (hunk 1 above)
Comment (tiationg-kho) [reply pkg/observability/opentelemetry_test.go]: Fixed.

File: pkg/observability/opentelemetry_test.go (hunk 2 above)
Comment (tiationg-kho) [reply pkg/observability/opentelemetry_test.go]: Fixed by renaming to responseRecorder.

File: pkg/observability/opentelemetry_test.go (hunk 3 above)
Comment (tiationg-kho) [reply pkg/observability/opentelemetry_test.go]: Fixed.

File: pkg/observability/opentelemetry_test.go (hunk 4 above)
Comment (tiationg-kho) [reply pkg/observability/opentelemetry_test.go:207]: Fixed by adding doc comment for example response body.

**Review by Lu-David — APPROVED** [review]

**Pull request merged**